### PR TITLE
Error changing parents

### PR DESCRIPTION
**Pull request by juliangruber**
This adds a failing test for changing the parent using `p2p.set()`, after a parent had already been set before.

The error message received is:

> UnhandledPromiseRejectionWarning: ValidationError: Message: Parents must be registered by at least one author | Code: parents_registered | Property: p2pcommons.parents

This currently blocks https://github.com/hypergraph-xyz/desktop/issues/69.

### CODE REVIEW

**Comment by jameslibscie** [comment]
So this turned out to be a mistake in the test combined with a bug in `init()` that failed to validate `parents`. Fixed `init()` and tests

File: tests/index.js [hunk 1]
```diff
@@ -753,32 +780,130 @@ test('follows: must not self-reference', async t => {
...
+    t.ok(
+      err instanceof SDK.errors.ValidationError,
+      'invalid parents should throw ValidationError'
+    )
+  }
+  await p2p.destroy()
+  t.end()
+})
+
+test.only('set: change versioned parent for unversioned registration', async t => {
```
Comment (dpaez) [inline tests/index.js:837]: remove `test.only`

**Comment by jameslibscie** [comment]
Forgot to push my last commit yesterday which removed the `.only` and expanded one of the tests to include a test for parent validation upon `init()`. There now :)

**Comment by jameslibscie** [comment]
@juliangruber Since this is your PR, I can't add you as a reviewer, but could you take a look? :)

**Comment by juliangruber** [comment]
Why are there unrelated changes in this PR again :/ could you please clean that up?

**Comment by juliangruber** [comment]
It looks like the changes merged in aren't part of the main branch yet, which is why this PR doesn't merge cleanly

**Comment by juliangruber** [comment]
It looks like https://github.com/p2pcommons/sdk-js/pull/210 was merged into this PR

**Comment by dpaez** [comment]
Hey, it looks like I added those changes. @jameslibscie I'm going to use your commits and create a new clean branch which your changes and @juliangruber test. We can discard this one. Sorry.

**Comment by juliangruber** [comment]
No worries!

**Pull request closed without merging**